### PR TITLE
update cilium and hubble-relay dockerfiles to use built-in buildx ARGs

### PR DESCRIPTION
**Pull request by xunholy**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Implementing docker buildx cross-architecture compilation for cilium and hubble-relay images.
Fixes: #issue-number

### CODE REVIEW

File: bpf/Makefile.bpf [hunk 1]
```diff
@@ -26,6 +26,13 @@ LLC   ?= llc
 HOST_CC    ?= gcc
 HOST_STRIP ?= strip
 
+TARGETARCH := $(shell go env GOARCH)
```
Comment (errordeveloper) [inline bpf/Makefile.bpf]: Please drop this line

File: bpf/Makefile.bpf [hunk 2]
```diff
@@ -26,6 +26,13 @@ LLC   ?= llc
...
+
+ifeq ($(TARGETARCH),arm64)
```
Comment (errordeveloper) [inline bpf/Makefile.bpf]: This needs to be more of a special case, it should be controlled only by a variable that is used inside of the Docker build. So if someone running on an Arm machine and trying to build it natively, it should use the native `gcc` and `strip` and not the cross-compiler.
```suggestion
ifeq ($(IMAGE_CROSS_TARGET_PLATFORM),linux/arm64)
```

File: images/cilium/Dockerfile [hunk 3]
```diff
@@ -6,41 +6,41 @@
...
+ENV GOOS=${TARGETOS} \
+  GOARCH=${TARGETARCH}
 
 RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
-  env GOARCH=arm64 CC=aarch64-linux-gnu-gcc \
-    make build-container install-container \
-      NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG RACE=$RACE PKG_BUILD=1 SKIP_DOCS=true DESTDIR=/out/linux/arm64 \
-      HOST_CC=aarch64-linux-gnu-gcc HOST_STRIP=aarch64-linux-gnu-strip
+  make build-container install-container \
+  NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG RACE=$RACE PKG_BUILD=1 SKIP_DOCS=true DESTDIR=/out/${TARGETOS}/${TARGETARCH}
```
Comment (errordeveloper) [inline images/cilium/Dockerfile]:
```suggestion
  NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG RACE=$RACE PKG_BUILD=1 SKIP_DOCS=true DESTDIR=/out/${TARGETOS}/${TARGETARCH} IMAGE_CROSS_TARGET_PLATFORM=${TARGETOS}/${TARGETARCH}
```
This is what I mean by setting an explicit variable for this...

**Review by errordeveloper — CHANGES_REQUESTED** [review]
Thanks for doing this @xUnholy! Just a few tweaks, and it should be good to go :)

File: bpf/Makefile.bpf (hunk 1 above)
Comment (xunholy) [reply bpf/Makefile.bpf]: done

File: bpf/Makefile.bpf (hunk 2 above)
Comment (xunholy) [reply bpf/Makefile.bpf]: What is going to set IMAGE_CROSS_TARGET_PLATFORM or are we expecting to update the Dockerfile `RUN` statement to set this as an `env`?

Comment (xunholy) [reply bpf/Makefile.bpf]: You answered question in the next feedback item :joy:

File: images/hubble-proto/Dockerfile [hunk 4]
```diff
@@ -13,25 +13,24 @@ RUN /tmp/install-protoplugins.sh
...
-COPY --from=builder /proto/protoc-gen-go /proto
-COPY --from=builder /proto/protoc-gen-go-grpc /proto
-COPY --from=builder /proto/protoc-gen-go-json /proto
-
-COPY Makefile.docker /proto/Makefile
+COPY --from=builder /proto/protoc-gen-go .
+COPY --from=builder /proto/protoc-gen-go-grpc .
+COPY --from=builder /proto/protoc-gen-go-json .
 
-WORKDIR /source
```
Comment (xunholy) [inline images/hubble-proto/Dockerfile:35]: Will removing this final WORKDIR make a difference in any subsequent images using this Dockerfile @errordeveloper? It appeared non-essential but probably good to double-check.

Comment (rolinh) [reply images/hubble-proto/Dockerfile:35]: The `hubble-proto` container is invoked by `api/v1/Makefile` this way:
```
docker container run --rm \
        --volume $(CURDIR):/source \
        cilium/hubble-proto:${version}
```
Therefore, removing `WORKDIR /source` will break API proto generation.

Comment (xunholy) [reply images/hubble-proto/Dockerfile:35]: Thanks, @Rolinh for confirming, I'll re-add this as it seemed suspicious that it was just arbitrarily added but then nothing in this Dockerfile suggested it was being used.

Comment (xunholy) [reply images/hubble-proto/Dockerfile:35]: reintroduced as mentioned above by @Rolinh it's required.

**Review by errordeveloper — APPROVED** [review]

**Pull request merged**